### DISCUSSION TITLE
Use oh-my-posh.exe from C:\Program Files (x86)\oh-my-posh\bin

### DISCUSSION THREAD

**Discussion by nzubair** [JanDeDobbeleer/oh-my-posh#1614] [Troubleshoot] [answered]
How can I force oh-my-posh module to use the oh-my-posh.exe from `C:\Program Files (x86)\oh-my-posh\bin` ?  

I have a need to have the oh-my-posh.exe installed for all users, which I was able to do by downloading and installing `install-amd64.exe` from the releases page. However, the module seems to have the executable path hard coded to `$HOME\.oh-my-posh` directory. Is there any way to override that and use the executable from Program Files?

**Comment by JanDeDobbeleer** [comment, marked as answer]
Remove the module and add the folder where you added oh-my-posh to your path. The module is only a wrapper around oh-my-posh, so you don't lose any functionality. Alternatively you can also install with winget/scoop as they add it to the path automatically.

**Comment by nzubair** [comment]
Thanks Jan. I've noticed a couple of issues so far. 

Using the All Users installer from releases page installs the files under `C:\Program Files (x86)\oh-my-posh`. However, it still creates the PATH values under "Current User" (HKCU) instead of Local Machine (HKLM).  I manually fixed the path, added the `C:\Program Files (x86)\oh-my-posh\themes` and `C:\Program Files (x86)\oh-my-posh\bin`  to system PATH. 

Removed and reinstalled the module.

Upon loading the first time, it still downloaded a copy of oh-my-posh.exe into   `$HOME\.oh-my-posh`  directory.  instead of using the one from `C:\Program Files (x86)\oh-my-posh`

As far as I can tell, the issue seems to be coming from `Get-PoshCommand` and `Get-PoshRootPath` functions.  `Get-PoshRootPath` sets up `POSH_PATH` to be `$HOME\.oh-my-posh` and then `Get-PoshCommand`  just joins `POSH_PATH` to executable name to come up with the full path.  At no point system PATH or `C:\Program Files (x86)` is referenced. 

Let me know if it is more appropriate to create an issue at this point.

**Reply by JanDeDobbeleer** [reply]
If you install oh-my-posh from winget/scoop/manual you need to remove the module. There's no need for it. The module only exists for those who only wish to install it solely for PowerShell. The installation methods shouldn't be mixed as yes, you end up with multiple binaries on different locations.

I'll check for the path on installation, that should be solvable.

**Reply by nzubair** [reply]
Oh, that clarifies a few things.  Removed the module, followed the instructions in the docs.

Thank you for the help.
